File: backend/internal/models/crop.go
package models

import "github.com/lib/pq"

type Crop struct {
	ID                    uint           `gorm:"primary_key" json:"id"`
	Name                  string         `json:"name"`
	Seasons               pq.StringArray `gorm:"type:text[]" json:"seasons"`
	TownRank              string         `json:"townRank"`
	GrowthTimeDay         int            `json:"growthTimeDay"`
	RegrowthTimeDay       int            `json:"regrowthTimeDay"`
	PossibleMaxHarvestDay int            `json:"possibleMaxHarvestDay"`
	ImageURL              string         `json:"imageURL"`
	SeedPrice             int            `json:"seedPrice"`
	PriceBase             int            `json:"buyPriceBase"`
	PriceBronze           int            `json:"buyPriceBronze"`
	PriceSilver           int            `json:"buyPriceSilver"`
	PriceGold             int            `json:"buyPriceGold"`
	PriceOsmium           int            `json:"buyPriceOsmium"`
}

File: frontend/src/types/crop.ts

export const TOWN_RANKS = [
    { label: "F", value: "F" },
    { label: "E", value: "E" },
    { label: "D", value: "D" },
    { label: "C", value: "C" },
    { label: "B", value: "B" },
    { label: "A", value: "A" },
    { label: "S", value: "S" },
]

export const SEASON = [
    { label: "Spring", value: "Spring" },
    { label: "Summer", value: "Summer" },
    { label: "Fall", value: "Fall" },
    { label: "Winter", value: "Winter" },
]

export type TownRankValue = (typeof TOWN_RANKS)[number]["value"];
export type SeasonValue = (typeof SEASON)[number]["value"]
export type Seasons = "Spring" | "Summer" | "Fall" | "Winter";


export interface Crop {
    id: number;
    name: string;
    seasons: Seasons;
    townRank: TownRankValue;
    growthTimeDay: number;
    regrowthTimeDay: number;
    possibleMaxHarvestDay: number;
    imageURL: string;
    buyPriceBase: number;
    buyPriceBronze: number;
    buyPriceSilver: number;
    buyPriceGold: number;
    buyPriceOsmium: number;
    seedPrice: number;
}